src/main/java/gov/cabinetoffice/gap/spotlight/publisher/service/RestService.java

```diff
@@ -0,0 +1,122 @@
+package gov.cabinetoffice.gap.spotlight.publisher.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+public class RestService {
```

Comment:
Should throw dedicated exceptions in this class too rather than generic ones

Reply:
Done